advisor/backend.go

```diff
@@ -188,6 +190,9 @@ type boltFinder struct {
 
 // Open the database for reading.
 func Open() (Finder, error) {
+	if !osutil.FileExists(dirs.SnapCommandsDB) {
+		return nil, os.ErrNotExist
+	}
 	db, err := bolt.Open(dirs.SnapCommandsDB, 0644, &bolt.Options{
```

Comment:
Silly `bolt.Open()` calling `os.OpenFile(.., os.O_RDONLY | os.O_CREATE)` even if ReadOnly mode is used.